### ISSUE TITLE
Oracle generates opposite parameter value for boolean types

### ISSUE THREAD

**Issue by jameschenjav** [juanluispaz/ts-sql-query#87]
[Source code:](https://github.com/juanluispaz/ts-sql-query/blob/2048ad85b67d30be3fdfede0a2d099ca325bc3f6/src/connections/AbstractOracleConnection.ts#L18)

```ts
    protected transformValueToDB(value: unknown, type: string): unknown {
        if (type === 'boolean' && typeof value === 'boolean') {
            if (value) {
                return 0
            } else {
                return 1
            }
        }
        return super.transformValueToDB(value, type)
    }
```

Obviously it converts `true` to `0` while `false` becomes `1`

**Comment by jameschenjav** [comment]
My case was simple, we use 'Y' and 'N' as boolean in Oracle. A very simple table just contains 2 columns: `id` and `active`, which is `VARCHAR(1)` and presents boolean type.

```ts
import { OracleDBPoolPromiseQueryRunner } from 'ts-sql-query/queryRunners/OracleDBPoolPromiseQueryRunner';
import { CustomBooleanTypeAdapter } from 'ts-sql-query/TypeAdapter';
import { Table } from 'ts-sql-query/Table';

class TableFoo extends Table<DBConnection, 'TableFoo'> {
  id = this.primaryKey('id', 'stringInt');

  isActive = this.column('active', 'boolean', new CustomBooleanTypeAdapter('Y', 'N'));

  constructor() {
    super('foo');
  }
}

const tFoo = new TableFoo();

const bar = async () => {
  const connection = new DBConnection(new OracleDBPoolPromiseQueryRunner(...));
  const select = connection
    .selectFrom(tFoo)
    .where(tFoo.isActive.equals(true))
    .select({
      id: tFoo.id,
      isActive: tFoo.isActive,
    });
  console.log({ query: select.query(), params: select.params() });
};
```

And the output is
```
{
  query: `select id as "id", case when active = 'Y' then 1 else 0 end as "isActive" from foo where case when active = 'Y' then 1 else 0 end = :0`,
  params: [ 0 ]
}
```

After I changed `tFoo.isActive.equals(true)` to `equals(false)`, the `params` became `1`, which was totally opposite.

**Comment by juanluispaz** [comment]
Released ts-sql-query 1.49.0 with this bug fix; sorry for the inconvenience.

Let me know if you need any support/help or if you want any new features.
